src/compas_ui/app.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from .session import Session
+
+
+class App(object):
+    """App singleton.
+
+    Parameters
+    ----------
+    scene : :class:`compas_ui.scene.Scene`, optional
+        The compas_ui scene object.
+    proxy : Union[:class:`compas_cloud.Proxy`, :class:`compas.rpc.Proxy`], optional
+        The compas rpc or compas_cloud Proxy object.
+
+    Attributes
+    ----------
+    scene : :class:`compas_ui.scene.Scene`
+        The compas_ui scene object.
+    proxy : Union[:class:`compas_cloud.Proxy`, :class:`compas.rpc.Proxy`]
+        The compas rpc or compas_cloud Proxy object.
+    session : :class:`compas_ui.session.Session`
+        The compas_ui session object.
+
+    """
+
+    _instance = None
+
+    def __new__(cls, *args, **kwargs):
+        if not cls._instance:
+            self = super(App, cls).__new__(cls)
+            self._initialized = False
+            cls._instance = self
+        return cls._instance
+
+    def __init__(self, scene=None, proxy=None):
+        if self._initialized:
+            return
+        self.scene = scene
+        self.proxy = proxy
+        self.session = Session()
```

Comment:
is it not simpler to set all these attributes as part of `__new__`?

Reply:
and with `Session` no longer a singleton i think this will not work, because `__init__` is called every time you do `Session()`...

Reply:
i guess the `if` in `__init__` prevents this, but that seems like a fragile mechanism to me...

Reply:
i would keep `Session` as a singleton as well. this allows it to be used independently from `App` if necessary...

Reply:
> is it not simpler to set all these attributes as part of `__new__`?

Somehow pylance does not recognize any parameters or attributes that are set in `__new__`, I think it only looks at `__init__`. That was the reason why I moved everything to init, unless we don't care about that?

Reply:
> i would keep `Session` as a singleton as well. this allows it to be used independently from `App` if necessary...

Sure that makes sense